IECoreAppleseed: small improvements

est77 opened the pull request:
In EditBlockHandler:

- Use smart pointers instead of raw pointers.
- Destroy the Renderer before the RendererController.

johnhaddon commented:
Does using auto_ptr mean that m_rendererController will be destroyed with delete rather than release()? If so, is that a problem? As you can see, I don't know too much about Appleseed's memory management, but I'm keen to be educated...

est77 commented:
In general, release() and appleseed's auto_release_ptr<> are important on windows, where depending on your compiler configuration, creating an object using new inside a dll and deleting it somewhere else can result in crashes. That's the reason all appleseed entities are created by factories and held by auto_release_ptrs. To make sure that new and delete are called from the appleseed dll.

In this particular case, the EditBlockHandler::RendererController is both created and deleted inside IECoreAppleseed, so it's fine. 

I need to check, but I think that RendererControllers don't really need a release() method at all. They are always created and deleted by clients of the appleseed lib.

est77 commented:
RendererController::release() was removed in appleseed.
https://github.com/appleseedhq/appleseed/commit/3bf80fe0307f08aad389df6efcb47cb02b3d93ca
